### PR TITLE
Log the missing `distributionGroupId` for the first session

### PR DESCRIPTION
**Pull request by roland-chernov-akvelon**
This PR addresses the issue with the first session log which is sent after app installation. The `distributionGroupId` field is missing in the first session log because it's sent before in-app updates are enabled. So this PR:

- Adds sending of distribution start session log after enabling in-app updates (first app launch after installation).
- Adds check to ignore distribution start session logs without session ID to handle cases when Analytics is disabled.
- Implements tests.

### CODE REVIEW

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/channel/DistributeInfoTracker.java [hunk 1]
```diff
@@ -26,14 +30,25 @@ public DistributeInfoTracker(String distributionGroupId) {
...
 
         /* Set current distribution group ID. */
         log.setDistributionGroupId(mDistributionGroupId);
     }
 
+    @Override
+    public boolean shouldFilter(@NonNull Log log) {
+
+        /* Filter out distribution start session log if session ID is not set (Analytics is disabled). */
+        if (log instanceof DistributionStartSessionLog && log.getSid() == null) {
```
Comment (guperrot) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/channel/DistributeInfoTracker.java]: This works but will be confusing for aria team as they will also receive this log but will be filtered out without them knowing.

Another solution to it sooner is to use SessionContext from core and check if the last session has a sessionId or null.

Comment (roland-chernov-akvelon) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/channel/DistributeInfoTracker.java]: Implemented using `SessionContext`. Thank you.

Please take a look.

File: sdk/appcenter-distribute/src/androidTest/java/com/microsoft/appcenter/distribute/DistributeSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            log.setTimestamp(new Date());
+            logs.add(log);
+        }
+        expectedContainer.setLogs(logs);
+        UUID sid = UUIDUtils.randomUUID();
+        for (Log log : logs) {
+            log.setSid(sid);
+        }
+
+        LogSerializer serializer = new DefaultLogSerializer();
```
Comment (guperrot) [inline sdk/appcenter-distribute/src/androidTest/java/com/microsoft/appcenter/distribute/DistributeSerializerTest.java]: Always add a comment after a blank line in code

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 3]
```diff
@@ -1768,4 +1801,31 @@ synchronized void setInstalling(ReleaseDetails releaseDetails) {
...
+
+        /*
+         * Session starts before in-app updates setup (using browser) so the first start session log
+         * is sent without distributionGroupId value.
+         *
+         * Send the distribution start session log if start session log without distributionGroupId
+         * value was sent before
+         */
+        SessionContext.SessionInfo lastSession = SessionContext.getInstance().getSessionAt(System.currentTimeMillis());
+        if (lastSession == null) {
```
Comment (guperrot) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java]: lastSession can have null sessionid if analytics disabled, you need to test that also (a launch session is created by the core before analytics adds one)

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeBeforeApiSuccessTest.java [hunk 4]
```diff
@@ -1498,4 +1501,40 @@ public boolean matches(Object argument) {
...
+        /* Enable in-app updates. */
+        start();
+        Distribute.getInstance().storeRedirectionParameters("r", "g", null);
+
+        /* Verify the log was sent. */
+        verify(mChannel).enqueue(any(DistributionStartSessionLog.class), eq(Distribute.getInstance().getGroupName()));
+    }
+
+    @Test
+    public void DontEnqueueDistributionStartSessionLogIfNoSessionsWereLoggedBefore() throws Exception {
```
Comment (guperrot) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeBeforeApiSuccessTest.java]: camelCase for method names, not PascalCase

File: sdk/appcenter-distribute/src/androidTest/java/com/microsoft/appcenter/distribute/DistributeSerializerTest.java (hunk 2 above)
Comment (roland-chernov-akvelon) [reply sdk/appcenter-distribute/src/androidTest/java/com/microsoft/appcenter/distribute/DistributeSerializerTest.java]: Added. Thank you.

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java (hunk 3 above)
Comment (roland-chernov-akvelon) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java]: Fixed. Thank you.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeBeforeApiSuccessTest.java (hunk 4 above)
Comment (roland-chernov-akvelon) [reply sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeBeforeApiSuccessTest.java]: Fixed. Also added a test to check for a `null` SID. Thank you.

**Comment by guperrot** [comment]
@roland-chernov-akvelon please avoid rewriting the history in PR as we do incremental reviews when possible. I have to review the entire changes since it was forced pushed.

**Comment by roland-chernov-akvelon** [comment]
@guperrot Okay, got it.
